index.html

```diff
@@ -3989,10 +3989,25 @@ <h3>JSON-LD</h3>
 
         <p>
 As elaborated upon in Section <a href="#json-processing"></a>, some software
-applications might not perform full JSON-LD processing.
-<a>Conforming document</a> authors are urged to not use JSON-LD keywords that
-affect document processing in the `@context` value, such as globally setting
-`@base` and `@language`, in order to increase interoperability.
+applications might not perform full JSON-LD processing. In order to increase
+interoperability, <a>conforming document</a> authors are urged to not use
+JSON-LD features that are not easily detected when not performing full
+JSON-LD processing. This guidance includes:
+        </p>
+
+        <ul>
+          <li>
+Not using JSON-LD keywords in the `@context` value that modify document
+processing, such as globally setting `@base` and `@language`, in order to
+increase interoperability.
+          </li>
+          <li>
+Always using the compact form for JSON-LD terms and types and always ensuring
+that those terms and types are defined in application-specific JSON-LD context
+files.
```

Comment:
```suggestion
Use of the compact form for JSON-LD terms and types and defining those
terms and types in application-specific JSON-LD context files
```

Reply:
For implementors doing JSON only processing, then having to define an application specific JSON-LD context file is perhaps asking too much, and they should be allowed to use URLs for the new application specific types

Reply:
Otherwise what is the point of having the '@vocab' keyword if every term and type must have a compact form defined in an `@context` file

Reply:
> For implementors doing JSON only processing, then having to define an application specific JSON-LD context file is perhaps asking too much, and they should be allowed to use URLs for the new application specific types

They don't have to do this because we now have an "issuer defined" `@vocab` value in the base context.

I'll try to figure out how to make this more clear -- remember, these are not normative MUSTs -- they are recommendations to increase interoperability. They don't /have to/ define "terms and types in application-specific JSON-LD context files" -- can you suggest some language that you'd be ok with?

Reply:
My comments here are relevant: https://github.com/w3c/vc-data-model/issues/1290#issuecomment-1737778877